subprojects/language-native/src/main/java/org/gradle/language/plugins/NativeBasePlugin.java

```diff
@@ -308,6 +308,15 @@ public String call() {
                                 return toolProvider.getSharedLibraryName("lib/" + names.getDirName() + library.getBaseName().get());
                             }
                         })));
+                        link.getInstallName().set(providers.provider(new Callable<String>() {
+                            @Override
+                            public String call() throws Exception {
```

Comment:
What about this instead:
```
link.getInstallName().set(link.getLinkedFile().map(new Transformer<String, RegularFile>() {
                            @Override
                            public String transform(RegularFile linkedFile) {
                                return linkedFile.getAsFile().getName();
                            }
                        }));
```
We're always setting the install name and let the args translator figure out if we actually need it or not.  It also avoids the ugly `project.provider {}` layer.